packages/cosmic-swingset/lib/ag-solo/vats/rendezvous.js

```diff
@@ -0,0 +1,248 @@
+// @ts-check
+import makeStore from '@agoric/store';
+
+import { assert, details } from '@agoric/assert';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+/**
+ * @typedef {string} PeerAddress the string representation of an Agoric peer's
+ * identity.  For Cosmos, this will be the Agoric account ID, like
+ * `agoric16vcjpzjr8uhvwwngyzkpcwe4x0qmnjdu63hc9u`.  For the simulated chain,
+ * this will be something like `mySimGCI-client`.
+ */
+
+/**
+ * @typedef {PeerAddress | Array<PeerAddress>} PeerAddresses one or many peer
+ * addresses
+ */
+
+const RENDEZVOUS_ALREADY = harden({
+  toString() {
+    return 'RENDEZVOUS_ALREADY';
+  },
+});
+
+/**
+ * @typedef {Object} RendezvousPrivateService A closely-held service associated
+ * with a specific Agoric peer.  It allows you to exchange values (possibly
+ * ocaps) with another peer, given a third-party that can exchange the other
+ * side's address strings between you.
+ * @property {() => string} getNamespaceName get the name of the rendezvous
+ * namespace.  This will typically be the chain identifier, like `testnet-4` for
+ * the Agoric testnet, or `mySim` for the simulated chain.  It is not
+ * necessarily unique.
+ * @property {() => PeerAddress} getLocalAddress get the address string for this
+ * private service
+ * @property {(peerAddrs: PeerAddresses, sendToPeer: any) => Rendezvous}
+ * startRendezvous create a new rendezvous instance for this local address,
+ * returning the value from the first matching peerAddress and send it
+ * `sendToPeer`
+ * @property {(peerAddrs: PeerAddresses, sendToPeer: any) =>
+ * any} completeRendezvous receive a value from the first matching peerAddress and
+ * send it `sendToPeer`, throwing an exception if no peer completed the
+ * rendezvous
```

Comment:
I think just from reading this, I would not understand what "returning the value from the first matching peerAddress and send it to `sendToPeer`" means. Can we explain more or differently?  The Rendezvous service is for sending a object reference to one or multiple peers, and each Rendezvous is only good for one such reference, right?

Reply:
I think I need more explanation/comments.

Reply:
> I think just from reading this, I would not understand what "returning the value from the first matching peerAddress and send it to `sendToPeer`" means. Can we explain more or differently?

"sending the `sendToPeer` value to the first matching peerAddress, and returning the value sent by that peer to the caller."?

> The Rendezvous service is for sending a object reference to one or multiple peers, and each Rendezvous is only good for one such reference, right?

The Rendezvous service is for finding the first peerAddress that matches with our own address, and exchanging values with it, (one in each direction).